### PR TITLE
Visual Studio /clr leads to linking errors

### PR DESCRIPTION
**Pull request by jmmorato**
Changes needed to avoid linking errors when compiling with C++/CLI.

Related Issue: #962

### CODE REVIEW

**Comment by mitza-oci** [comment]
Is there a change you can make to azure-pipelines.yml to test this?

If we had that set up we could detect this problem before a release.  Though it would be much better to have all supported compilers+options understand the syntax of standard C++ (98, not even new stuff), I guess there's not much we can do in this case.

**Comment by jmmorato** [comment]
We could implement in a separated repository the example explained on https://objectcomputing.com/resources/publications/mnb/using-tao-and-opendds-with-net-part-1 and try to build it at the end of the Visual Studio pipelines.

File: dds/DCPS/unique_ptr.h [hunk 1]
```diff
@@ -132,7 +132,7 @@ class unique_ptr {
   }
 
 private:
-  unique_ptr(const unique_ptr&);
+  unique_ptr(const unique_ptr&) {};
```
Comment (mitza-oci) [inline dds/DCPS/unique_ptr.h]: `unique_ptr` is not copyable, so the copy constructor is deliberately declared but not defined.

Comment (jmmorato) [reply dds/DCPS/unique_ptr.h]: Seems that the modification is not needed anymore to link C++/CLI projects. I would say it was fixed on this commit 8b02ca2

**Pull request merged**